iotedgedev/cli.py

```diff
@@ -663,14 +670,18 @@ def list_subscriptions_and_set_default():
     if not azure_cli.list_subscriptions():
         sys.exit()
     default_subscriptionId = azure_cli.get_default_subscription()
-    return default_subscriptionId
+    output.info(f"Select an Azure Subscription Name or Id [{default_subscriptionId}]:")
+    return input() or default_subscriptionId
 
 
-def header_and_default(header, default, default2=None):
+def header_and_default(header, prompt, default, default2=None, options=None):
     output.header(header)
     if default == '' and default2 is not None:
         return default2
```

Comment:
>return default2 [](http://example.com/codeflow?start=8&length=15)

if the code returns here it will miss the input prompt. Let's use an intermediate variable to capture the return value

Reply:
I think the `default2` is confusing, and is only used at line 729. I think it may be better to remove the `default2` argument, and instead change line 729 to something like
`default=lambda: header_and_default('RESOURCE GROUP LOCATION', 'Enter a Resource Group Location', envvars.RESOURCE_GROUP_LOCATION or 'westus',` so it chooses `westus` as default if the environment variable is empty, which I assume is the goal of `default2`

wdyt?

Reply:
your proposal to update l 729 sounds good to me

Reply:
fixed in https://github.com/Azure/iotedgedev/pull/585/commits/3b9a769a647e2c14096c778d81e887765704bc90